lib/utensils/have_hash_matcher.rb

```diff
@@ -0,0 +1,60 @@
+RSpec::Matchers.define :have_hash do |expected|
+  match do |array|
+    synchronize do
+      hash = find_hash(array, expected)
+
+      expect(hash).to include(expected)
+      expect(hash).to_not include(@unexpected_keys)
+    end
+  end
+
+  chain :without do |*unexpected_keys|
+    @unexpected_keys = unexpected_keys
+  end
+
+  failure_message do
+    @failure_message
+  end
+
+  private
+
+    def find_hash(array, expected)
+      hash = array.reverse.find do |item|
```

Comment:
Could you please explain why we reverse the array here?

Reply:
Honestly I have no idea 😅. This is copied from https://github.com/cookpad/streamy, the current source of the matcher... and a quick git trawl isn't really helping me understand this line 🤔

Reply:
Could it be that if we are writing an integration spec and include a check for an activity log after several of the steps in the spec, the log entry that we are after is more likely to be at or close to the end of the array?